build/debug_keybase/main.go

```diff
@@ -0,0 +1,197 @@
+	for _, privHexString := range validatorKeysPairMap {
+		if _, err := limit.Execute(func() {
+			// Import the keys into the keybase with no passphrase or hint as these are for debug purposes
+			keyPair, err := cryptoPocket.CreateNewKeyFromString(privHexString, "", "")
+			if err != nil {
+				errCh <- err
+				return
+			}
+
+			// Use key address as key in DB
+			addrKey := keyPair.GetAddressBytes()
+
+			// Encode KeyPair into []byte for value
+			keypairBz, err := keyPair.Marshal()
+			if err != nil {
+				errCh <- err
+				return
+			}
+			if err := wb.Set(addrKey, keypairBz); err != nil {
+				errCh <- err
+				return
+			}
+		}); err != nil {
+			panic(err)
+		}
+	}
+
+	if err := limit.WaitAndClose(); err != nil {
+		panic(err)
+	}
+
+	// Check if any goroutines returned an error
+	select {
+	case err := <-errCh:
+		panic(err)
+	default:
+	}
+
+	if err := wb.Flush(); err != nil {
+		panic(err)
+	}
+
+	fmt.Println("✅ Keybase initialized!")
+
+	fmt.Println("⚙️ Creating a dump of the Keybase...")
+	backupFile, err := os.Create(targetFilePath)
+	if err != nil {
+		panic(err)
+	}
+	defer backupFile.Close()
+	if _, err := db.Backup(backupFile, 0); err != nil {
+		panic(err)
+	}
+
+	// Close DB connection
+	if err := kb.Stop(); err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("✅ Keybase dumped in %s\n", targetFilePath)
+}
+
+func parseValidatorPrivateKeysFromEmbeddedYaml(privateKeysYamlBytes []byte) (map[string]string, error) {
+	// Parse the YAML file and load into the config struct
+	var config struct {
+		ApiVersion string            `yaml:"apiVersion"`
+		Kind       string            `yaml:"kind"`
+		MetaData   map[string]string `yaml:"metadata"`
+		Type       string            `yaml:"type"`
+		StringData map[string]string `yaml:"stringData"`
+	}
+	if err := yaml.Unmarshal(privateKeysYamlBytes, &config); err != nil {
+		return nil, err
+	}
+	validatorKeysMap := make(map[string]string)
+
+	for id, privHexString := range config.StringData {
+		validatorKeysMap[id] = privHexString
+	}
+	return validatorKeysMap, nil
+}
+
+func cleanupStaleFiles(targetFolderPath string) {
+	fmt.Printf("🧹 Cleaning up stale backup files in in %s\n", targetFolderPath)
+
+	if err := filepath.Walk(targetFolderPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			panic(err)
+		}
+
+		if !info.IsDir() && (filepath.Ext(path) == ".bak" || filepath.Ext(path) == ".md5") {
+			if err := os.Remove(path); err != nil {
+				panic(err)
+			}
+			fmt.Println("🚮 Deleted file:", path)
+		}
+
+		return nil
+	}); err != nil {
+		panic(err)
+	}
+}
+
+func createHashFile(hashFilePath string) {
+	fmt.Printf("🔖 Creating the MD5 hash file used to track consistency: %s\n", hashFilePath)
+
+	file, err := os.Create(hashFilePath)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+	if _, err := file.WriteString("This file is used to check if the keybase dump is in sync with the YAML file. Its name is the MD5 hash of the private_keys.yaml"); err != nil {
```

Comment:
I ❤️  this pattern: the name being the hash and the content being the semantic explanation.

💯